model/project_parser_s3.go

```diff
@@ -79,9 +80,18 @@ func (s *ParserProjectS3Storage) UpsertOne(ctx context.Context, pp *ParserProjec
 	if err != nil {
 		return errors.Wrap(err, "getting config")
 	}
+	isDegradedMode := !config.ServiceFlags.CPUDegradedModeDisabled
 	maxSize := config.TaskLimits.MaxParserProjectSize
-	if maxSize > 0 && parserProjectLen > maxSize {
-		return errors.Errorf("parser project exceeds the system limit (%v > %v bytes).", parserProjectLen, maxSize)
+	if isDegradedMode {
+		maxSize = config.TaskLimits.MaxDegradedModeParserProjectSize
+	}
+	if maxSize > 0 && parserProjectLen > maxSize*1024*1024 {
```

Comment:
Could you comment an explanation about the 1024*1024 factor? (Ideally that could be a constant also)